lib/elixir/lib/calendar/duration.ex

```diff
@@ -5,10 +5,77 @@ defmodule Duration do
   A `Duration` struct represents a collection of time scale units,
   allowing for manipulation and calculation of durations.
 
-  Date and time scale units are represented as integers, allowing for both positive and negative values.
+  Date and time scale units are represented as integers, allowing for
+  both positive and negative values.
 
-  Microseconds are represented using a tuple `{microsecond, precision}`. This ensures compatibility with
-  other calendar types implementing time, such as `Time`, `DateTime`, and `NaiveDateTime`.
+  Microseconds are represented using a tuple `{microsecond, precision}`.
+  This ensures compatibility with other calendar types implementing time,
+  such as `Time`, `DateTime`, and `NaiveDateTime`.
+
+  ## Shifting
+
+  The most common use of durations in Elixir's standard library is to
+  "shift" the calendar types.
+
+      iex> Date.shift(~D[2016-01-03], month: 2)
+      ~D[2016-03-03]
+
+  In the example above, `Date.shift/2` automatically converts the units
+  into a Duration struct, although one can also be given directly:
+
+      iex> Date.shift(~D[2016-01-03], Duration.new!(month: 2))
+      ~D[2016-03-03]
+
+  It is important to notice that shifting is not an arithmetic operation
```

Comment:
It would be helpful to clarify this one a bit more, as its not clear why it yields to different dates. Possibly showing the intermediate date on the first example would be helpful

Reply:
Done!